volumes/miovision/sql/function/function-aggregate-volumes_15min_mvt.sql

```diff
@@ -15,70 +15,96 @@ DECLARE
-        --remove the 15 minute bin containing any unacceptable gaps
-        AND dt.datetime_bin = un.datetime_bin
-    --To get 1min bins
-    LEFT JOIN miovision_api.volumes AS v ON
-        --help query choose correct partition
+    WHERE
+        --0 padding for certain modes (padding)
+        im.classification_uid IN (1,2,6,10)
+        AND im.intersection_uid = ANY(target_intersections)
+        
+    UNION ALL
+    
+    --real volumes
+    SELECT
+        v.intersection_uid,
+        datetime_bin_15(v.datetime_bin) AS datetime_bin,
+        v.classification_uid,
+        v.leg,
+        v.movement_uid,
+        SUM(volume)
+    FROM miovision_api.volumes AS v
+    --only aggregate common movements
+    JOIN miovision_api.intersection_movements USING (
+        intersection_uid, classification_uid, leg, movement_uid
+    )
+    WHERE
         v.datetime_bin >= start_date
         AND v.datetime_bin < end_date
-        AND v.datetime_bin >= dt.datetime_bin
-        AND v.datetime_bin < dt.datetime_bin + interval '15 minutes'
-        AND v.intersection_uid = im.intersection_uid
-        AND v.classification_uid = im.classification_uid
-        AND v.leg = im.leg
-        AND v.movement_uid = im.movement_uid
+        AND v.intersection_uid = ANY(target_intersections)
+    GROUP BY
+        v.intersection_uid,
+        datetime_bin_15(v.datetime_bin),
+        v.classification_uid,
+        v.leg,
+        v.movement_uid
+),
+
+aggregate_insert AS (
+    INSERT INTO miovision_api.volumes_15min_mvt(
+        intersection_uid, datetime_bin, classification_uid, leg, movement_uid, volume
+    )
+    SELECT DISTINCT ON (
+        v.intersection_uid, v.datetime_bin, v.classification_uid, v.leg, v.movement_uid
+    )
+        v.intersection_uid,
+        v.datetime_bin,
+        v.classification_uid,
+        v.leg,
+        v.movement_uid,
+        CASE
+            --set unacceptable gaps as nulls
+            WHEN un.datetime_bin IS NOT NULL THEN NULL
+            --gap fill with zeros (restricted to certain modes in temp CTE)
+            ELSE v.volume
+        END AS volume
+    FROM temp AS v
+    JOIN miovision_api.intersections AS i USING (intersection_uid)
+    --set unacceptable gaps as null
+    LEFT JOIN miovision_api.unacceptable_gaps AS un USING (
+        intersection_uid, datetime_bin
+    )
     WHERE
         -- Only include dates during which intersection is active 
         -- (excludes entire day it was added/removed)
-        dt.datetime_bin > mai.date_installed + interval '1 day'
+        v.datetime_bin >= i.date_installed + interval '1 day'
         AND (
-            mai.date_decommissioned IS NULL
-            OR (dt.datetime_bin < mai.date_decommissioned - interval '1 day')
+            i.date_decommissioned IS NULL
+            OR (v.datetime_bin < i.date_decommissioned - interval '1 day')
         )
-        --exclude movements already aggregated
-        AND v.volume_15min_mvt_uid IS NULL
```

Comment:
is this excluded cause we have the clearing function?

Reply:
yes exactly! it is redundant - we always run clear + aggregate together whether through run-api or airflow.